test/e2e/far_e2e_test.go

```diff
@@ -99,12 +100,29 @@ var _ = Describe("FAR E2e", func() {
 			// save the node's boot time prior to the fence agent call
 			nodeBootTimeBefore, err = e2eUtils.GetBootTime(clientSet, nodeName, testNsName, log)
 			Expect(err).ToNot(HaveOccurred(), "failed to get boot time of the node")
+
+			// set the node as "unhealthy" by disabling kubelet
+			makeNodeUnready(selectedNode)
+
 			far := createFAR(nodeName, fenceAgent, testShareParam, testNodeParam)
 			DeferCleanup(deleteFAR, far)
 		})
+		When("running FAR to reboot a node", func() {
+			It("should successfully remediate it", func() {
+				checkRemediation(nodeName, succeesRebootMessage, nodeBootTimeBefore)
+			})
+		})
+
 		When("running FAR to reboot two nodes", func() {
 			It("should successfully remediate the first node", func() {
 				checkRemediation(nodeName, succeesRebootMessage, nodeBootTimeBefore)
+				// run the second remediation on a different node
```

Comment:
This looks like setup code for the next test, it shouldn't be here.
I think all this node selection stuff can and should be in the `BeforeEach()` block above...

Reply:
I think it suits better an AfterEach, so I have used it.